pkg/vmprovider/providers/vsphere/storage/storageclass.go

```diff
@@ -20,17 +24,36 @@ func GetStoragePolicyID(
 	client ctrlclient.Client,
 	storageClassName string) (string, error) {
 
-	sc := &storagev1.StorageClass{}
-	if err := client.Get(vmCtx, ctrlclient.ObjectKey{Name: storageClassName}, sc); err != nil {
-		vmCtx.Logger.Error(err, "Failed to get StorageClass", "storageClass", storageClassName)
+	var policyID string
+
+	zones, err := topology.GetAvailabilityZones(vmCtx, client)
+	if err != nil {
 		return "", err
 	}
 
-	policyID, ok := sc.Parameters["storagePolicyID"]
-	if !ok {
-		return "", fmt.Errorf("StorageClass %s does not have 'storagePolicyID' parameter", storageClassName)
-	}
+	if pkgconfig.FromContext(vmCtx).Features.PodVMOnStretchedSupervisor && len(zones) > 1 {
```

Comment:
Here and elsewhere can we change things around to only grab the AZs when PodVMOnStretchedSupervisor is enabled?